src/d3a/models/myco_matcher/myco_external_matcher.py

```diff
@@ -162,45 +172,69 @@ def _get_bids_offers(market: TwoSidedMarket, filters: Dict):
                 offer.serializable_dict() for offer in offers.values())
         return bids_list, offers_list
 
-    def _get_validated_recommendations(
-            self, recommendations: List[Dict]) -> List[BidOfferMatch.serializable_dict]:
-        """Return a validated list of BidOfferMatch instances."""
-        validated_recommendations = []
+
+class MycoExternalMatcherValidator:
+    """Class responsible for the validation of external recommendations."""
+
+    # Blocking exceptions are ones that should block the matching process
+    BLOCKING_EXCEPTIONS = (MycoValidationException, )
+
+    @staticmethod
+    def validate_valid_dict(matcher: MycoExternalMatcher, recommendation: Dict):
+        """Check whether the recommendation dict is valid."""
+        if not BidOfferMatch.is_valid_dict(recommendation):
+            raise MycoValidationException(f"BidOfferMatch is not valid {recommendation}")
+
+    @staticmethod
+    def validate_market_exists(matcher: MycoExternalMatcher, recommendation: Dict):
+        """Check whether myco matcher is keeping track of the received market id"""
+        market = matcher.markets_mapping.get(recommendation.get("market_id"))
+        if market is None:
+            # The market doesn't exist
+            raise MycoValidationException(
+                f"Market with id {recommendation.get('market_id')} doesn't exist."
+                f"{recommendation}")
+
+    @staticmethod
+    def validate_orders_exist_in_market(matcher: MycoExternalMatcher, recommendation: Dict):
+        """Check whether all bids/offers exist in the market."""
+
+        market = matcher.markets_mapping.get(recommendation.get("market_id"))
+        market_offers = [
+            market.offers.get(offer["id"]) for offer in recommendation["offers"]]
+        market_bids = [market.bids.get(bid["id"]) for bid in recommendation["bids"]]
+
+        if not (all(market_offers) and all(market_bids)):
+            # If not all offers bids exist in the market, skip the current recommendation
+            raise InvalidBidOfferPairException(
+                "Not all bids and offers exist in the market.")
+
+    @classmethod
+    def _validate(cls, matcher: MycoExternalMatcher, recommendation: Dict):
+        """Call corresponding validation methods."""
+        cls.validate_valid_dict(matcher, recommendation)
+        cls.validate_market_exists(matcher, recommendation)
+        cls.validate_orders_exist_in_market(matcher, recommendation)
+
+    @classmethod
+    def validate_and_report(cls, matcher: MycoExternalMatcher, recommendations: Dict) -> Dict:
```

Comment:
I see your concern, This can be done indeed, however I see that these two classes are tightly coupled (This validator class will only work with the External Matcher)
So I propose either keeping them both in the same module, or creating a package (I also think that this should be avoided now but can be done if we planned to expand this external matcher functionality to add more classes)
In any case, separating them into 2 modules may cause some circular imports issues (The matcher imports the validator and the validator imports the matcher for type hints, this can be silenced though)
So it's your call, thanks!